test/conformance/ingress/util.go

```diff
@@ -54,6 +54,19 @@ import (
 
 var rootCAs = x509.NewCertPool()
 
+// TestRoundTripper wraps the given http.RoundTripper and
+// sets a custom UserAgent.
+type TestRoundTripper struct {
```

Comment:
neither this type, not its members have to be public.
Also next can be anonymized, but up to you.

Reply:
> neither this type, not its members have to be public.

I made the type private, but kept the members public since they are part of the API for this type, not private implementation details.

> Also next can be anonymized, but up to you.

My preference is to be explicit here.

Reply:
> I made the type private, but kept the members public since they are part of the API for this type, not private implementation details.

That's not really how Go works, right?  The whole type is implementation detail.

Reply:
Maybe we are talking at cross purposes here. If I do this:

```Go
u := uaRoundTripper {
        next: http.DefaultTransport,
        userAgent: "foo",
}
```

That would work because I'm only using uaRoundTripper from inside this package. By making the members "public", I'm hinting at their purpose, even though the publicness isn't enforced within the package. But it's not a big deal, and I can just flip them :)